src/CTA.WebForms/Helpers/TagConversion/TagTemplateParser.cs

```diff
@@ -1,9 +1,211 @@
-﻿namespace CTA.WebForms.Helpers.TagConversion
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using CTA.Rules.Config;
+using CTA.WebForms.Services;
+using HtmlAgilityPack;
+
+namespace CTA.WebForms.Helpers.TagConversion
 {
     /// <summary>
     /// Parses tag conversion templates into a usable format.
     /// </summary>
     public class TagTemplateParser
     {
+        private CodeBehindReferenceLinkerService _codeBehindLinkerService;
+
+        public const string TargetAttributeGroup = "TargetAttribute";
+        public const string SourceAttributeGroup = "SourceAttribute";
+        public const string Context0Group = "Context0";
+        public const string Context1Group = "Context1";
+
+        /// <summary>
+        /// Regular expression that matches attribute assignments with template replacement for values.
```

Comment:
Can we have sample of valid matches and a good explanation of what the regex string is looking for exactly?